File: Task-03/Java/subtask3.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);    // inputing values 
        System.out.print("Enter number- ");  
        int n = sc.nextInt(); 

        // Upper half 
        for (int i = 0; i < n; i += 2) {
            for (int j = n - 1; j > i; j--) {
                System.out.print(" ");
            }
            for (int k = 0; k <= i; k++) {
                System.out.print("* ");
            }
            System.out.println();
        }

        // Lower half
        for (int i = n - 3; i >= 0; i -= 2) {
            // Print spaces
            for (int j = n - 1; j > i; j--) {
                System.out.print(" ");
            }
            for (int k = 0; k <= i; k++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }
}

File: Task-03/Rust/subtask3.rs
use std::io;

fn main() {
    println!("Enter n value: ");
    
    // Read user input
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");
    
    // input as an integer
    let n: usize = input.trim().parse().expect("Please enter a valid number");

    // Generate the upper part o
    for i in 1..=n {
        if i % 2 == 1 { // Only for odd numbers
            let spaces = (n - i) / 2;
            let stars = i;
            println!("{}{}", " ".repeat(spaces), "*".repeat(stars));
        }
    }

    // Generate the lower part 
    for i in (1..n).rev() {
        if i % 2 == 1 { // Only for odd numbers
            let spaces = (n - i) / 2;
            let stars = i;
            println!("{}{}", " ".repeat(spaces), "*".repeat(stars));
        }
    }
}
